nsqdserver/http.go

```diff
@@ -279,30 +286,68 @@ func (s *httpServer) internalPUB(w http.ResponseWriter, req *http.Request, ps ht
 
 	if s.ctx.checkForMasterWrite(topic.GetTopicName(), topic.GetTopicPart()) {
 		var err error
-		traceIDStr := params.Get("trace_id")
```

Comment:
I see, will apply needTraceRsp to decide whether write trace info in resp or not.